File: src/main/java/org/stevenlowes/tools/lifx_controller/Messages/Light/SetInfrared.java
package org.stevenlowes.tools.lifx_controller.Messages.Light;

import org.stevenlowes.tools.lifx_controller.LifxCommander.CommonMethods;
import org.stevenlowes.tools.lifx_controller.Messages.DataTypes.Payload;

public class SetInfrared extends Payload {
	private final int code = 122;
	private int brightness;		// 16-Bits (Unsigned)
	
	public SetInfrared() {
		brightness = 0;
	}
	
	public SetInfrared(int brightness) {
		this.brightness = brightness;
	}
	
	public SetInfrared(SetInfrared setInfrared) {
		brightness = setInfrared.brightness;
	}
	
	public int getCode() {
		return code;
	}
	
	public int getBrightness() {
		return brightness;
	}
	
	public void setBrightness(int brightness) {
		this.brightness = brightness;
	}
	
	public byte[] getByteArray() {
		byte[] byteArray = new byte[2];
		
		String brightnessBinStr = String.format("%16s", Integer.toBinaryString(brightness)).replace(' ', '0');
		byteArray = CommonMethods.convertBinaryStringToLittleEndianByteArray(brightnessBinStr);
		
		return byteArray;
	}
	
}

File: src/main/java/org/stevenlowes/tools/lifxcontroller/messages/light/SetInfrared.kt
package org.stevenlowes.tools.lifxcontroller.messages.light

import org.stevenlowes.tools.lifxcontroller.CommonMethods
import org.stevenlowes.tools.lifxcontroller.messages.datatypes.payloads.CustomReadPayload

class SetInfrared(var brightness: Int = 0) : CustomReadPayload(122) {
    override val byteArray: ByteArray
        get() {
            val brightnessBinStr = String.format("%16s", Integer.toBinaryString(brightness)).replace(' ', '0')
            return CommonMethods.convertBinaryStringToLittleEndianByteArray(brightnessBinStr)
        }
}
